File: src/main/java/um/tesoreria/facturador/model/dto/ComprobanteDto.java
package um.tesoreria.facturador.model.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import um.tesoreria.facturador.tool.Jsonifier;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class ComprobanteDto {

    private Integer comprobanteId;
    private String descripcion = "";
    private Integer tipoTransaccionId;
    private Byte ordenPago = 0;
    private Byte aplicaPendiente = 0;
    private Byte cuentaCorriente = 0;
    private Byte debita = 0;
    private Long diasVigencia = 0L;
    private Byte facturacionElectronica = 0;
    private Integer comprobanteAfipId;
    private Integer puntoVenta;
    private String letraComprobante;
    private ComprobanteAfipDto comprobanteAfip;

    public String jsonify() {
        return Jsonifier.builder(this).build();
    }

}

File: src/main/java/um/tesoreria/facturador/kotlin/tesoreria/core/dto/ComprobanteDto.kt
package um.tesoreria.facturador.kotlin.tesoreria.core.dto

import com.fasterxml.jackson.core.JsonProcessingException
import com.fasterxml.jackson.databind.json.JsonMapper

data class ComprobanteDto(

    var comprobanteId: Int? = null,
    var descripcion: String = "",
    var tipoTransaccionId: Int? = null,
    var ordenPago: Byte = 0,
    var aplicaPendiente: Byte = 0,
    var cuentaCorriente: Byte = 0,
    var debita: Byte = 0,
    var diasVigencia: Long = 0,
    var facturacionElectronica: Byte = 0,
    var comprobanteAfipId: Int? = null,
    var puntoVenta: Int? = null,
    var letraComprobante: String? = null,
    var comprobanteAfip: ComprobanteAfipDto? = null,

    ) {

    fun jsonify(): String {
        try {
            return JsonMapper
                .builder()
                .findAndAddModules()
                .build()
                .writerWithDefaultPrettyPrinter()
                .writeValueAsString(this)
        } catch (e: JsonProcessingException) {
            return "jsonify error ${e.message}"
        }
    }

}
